src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaClientProduceFactory.java

```diff
@@ -1413,15 +1415,30 @@ private void encode(
             int progress,
             int limit)
         {
+            final int length = limit - progress;
+            if (encodeSlot != NO_SLOT &&
+                encodeableRequestBytes > 0 &&
+                encodeSlotLimit + length + KAFKA_RECORD_FRAMING + encodeableRecordHeadersBytes > encodePool.slotCapacity())
+            {
+                doNetworkData(traceId, EMPTY_BUFFER, 0, 0);
+            }
+
             KafkaProduceClientEncoder previous = null;
             dataFlags = flags & FLAGS_INIT;
+            final int previousProgress = progress;
 
             while (progress <= limit && previous != encoder)
             {
                 previous = encoder;
                 progress = encoder.encode(this, traceId, authorization, budgetId, reserved, flags, payload, extension,
                     progress, limit);
             }
+
+            final int credit = progress - previousProgress;
+            if (credit > KAFKA_RECORD_FRAMING)
```

Comment:
Is this comparison with `KAFKA_RECORD_FRAMING` necessary or can it be changed to comparison with `0`?

Reply:
I just thought that we don't want to send small credit.